src/components/Modal.js

```diff
@@ -80,6 +78,11 @@ class Modal extends Component {
 
         if (this.props.isOpen && !this.props.isStatic) {
             if (this.props.onCancel) {
+
+                // Ensure that onToggle gets called.
+                this.onToggle(false, this.props);
+
+                // Call the user supplied function.
```

Comment:
The problem turned out to be a bug in the parent class which was not calling the componentWillUnmount on the child class. This is fixed and the PR was updated.